packages/icons-react/types/Icon.d.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright IBM Corp. 2019, 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import * as React from 'react';
+
+type IconSize = 16 | 20 | 24 | 32 | '16' | '20' | '24' | '32' | number | string;
```

Comment:
[There's an outstanding comment](https://github.com/carbon-design-system/carbon/pull/11533/files#r890708661) relating to this that I think needs some discussion. You can ignore my comment on it because the answer to what I asked over there is "yes", but the comment about IDE autocompletion is worth exploring further.

Also, I'm not sure why strings are allowable on this prop. @tay1orjones Do react icons allow you to specify string versions of the number sizes?

Reply:
@jdharvey-ibm Yes, both strings or number literals [are allowed](https://stackblitz.com/edit/github-b36b4u?file=src%2FApp.jsx)